app/models/physical_server_profile.rb

```diff
@@ -0,0 +1,29 @@
+class PhysicalServerProfile < ApplicationRecord
+  acts_as_miq_taggable
+
+  include NewWithTypeStiMixin
+  include TenantIdentityMixin
+  include SupportsFeatureMixin
+  include EventMixin
+  include ProviderObjectMixin
+  include EmsRefreshMixin
+
+  include_concern 'Operations'
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :physical_server_profiles,
+    :class_name => "ManageIQ::Providers::PhysicalInfraManager"
+
+  belongs_to :assigned_server, :optional => true, :class_name => "ManageIQ::Providers::PhysicalInfraManager::PhysicalServer"
+  belongs_to :associated_server, :optional => true, :class_name => "ManageIQ::Providers::PhysicalInfraManager::PhysicalServer"
+
+  delegate :queue_name_for_ems_operations, :to => :ext_management_system, :allow_nil => true
+
+  def my_zone
+    ems = ext_management_system
+    ems ? ems.my_zone : MiqServer.my_zone
+  end
+
+  def event_where_clause(assoc = :ems_events)
+    ["#{events_table_name(assoc)}.physical_server_profile_id = ?", id]
```

Comment:
In order for this to work you'd have to add a `t.references :physical_server_profile` to `event_streams` table, which we can do but maybe we can save that for later once we actually have events being caught for physical_server_profiles?

Reply:
I suggest we save for later. Would you prefer to remove this completely or keep it for later?

Reply:
Best to remove completely until the new column is ready I think

Reply:
Done.